### PR TITLE
Added TabularRegressionData extending TabularData

### LINKED ISSUE #527: Create TabularRegressionData DataModule

### PR DESCRIPTION
**Pull request by sumanmichael**
## What does this PR do?
**Created TabularRegressionData DataModule**

TabularRegressionData has a lot common with TabularData in tabular/classification/data.py.
Here's what I did.
- moved tabular/classification/data.py -> tabular/data.py
- created TabularClassificationData in new tabular/classification/data.py & TabularRegressionData in new tabular/regression/data.py extending the TabularData in tabular/data.py by fixing the respective is_regression bool param.
- changed required __ init __.py files

Fixes #527 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/tabular/regression/data.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    from pandas.core.frame import DataFrame
+else:
+    DataFrame = object
+
+from flash.tabular.data import TabularData
+
+
+class TabularRegressionData(TabularData):
+
+    @classmethod
```
Comment (tchaton) [inline flash/tabular/regression/data.py:32]:
```suggestion
class TabularRegressionData(TabularData):

    is_regression: True

    @classmethod

```

File: flash/tabular/data.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    def uncollate(self, batch: Any) -> Any:
+        return batch
+
+
+class TabularData(DataModule):
+    """Data module for tabular tasks"""
+
+    preprocess_cls = TabularPreprocess
+    postprocess_cls = TabularPostprocess
+
```
Comment (tchaton) [inline flash/tabular/data.py:247]:
```suggestion
class TabularData(DataModule):
    """Data module for tabular tasks"""

    preprocess_cls = TabularPreprocess
    postprocess_cls = TabularPostprocess
    
    is_regression: Optional[bool] = None

```

File: flash/tabular/classification/data.py [hunk 3]
```diff
@@ -11,318 +11,23 @@
...
-
-        if data_frames[0][target_fields].dtype == object:
-            # if the target_fields is a category, not an int
-            target_codes = _generate_codes(data_frames, [target_fields])
-        else:
-            target_codes = None
-        codes = _generate_codes(data_frames, categorical_fields)
 
-        return mean, std, classes, codes, target_codes
+class TabularClassificationData(TabularData):
```
Comment (tchaton) [inline flash/tabular/classification/data.py:30]:
```suggestion
class TabularClassificationData(TabularData):

       is_regression = False
```

File: flash/tabular/regression/data.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    def from_data_frame(
```
Comment (tchaton) [inline flash/tabular/regression/data.py:33]: This would enable us to duplicate their classmethod as they would live on the TabularData class.

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley:

**Pull request merged**